CRM/Pledge/Page/UserDashboard.php

```diff
@@ -35,25 +35,20 @@ public function listPledges() {
     $controller->process();
     $controller->run();
 
-    // add honor block.
-    $honorParams = [];
+    // Add honor block.
     $honorParams = CRM_Pledge_BAO_Pledge::getHonorContacts($this->_contactId);
-    if (!empty($honorParams)) {
-      // assign vars to templates
-      $this->assign('pledgeHonorRows', $honorParams);
-      $this->assign('pledgeHonor', TRUE);
-    }
-    $session = CRM_Core_Session::singleton();
-    $loggedUserID = $session->get('userID');
-    $this->assign('loggedUserID', $loggedUserID);
+    $this->assign('pledgeHonorRows', $honorParams);
+    $this->assign('pledgeHonor', !empty($honorParams));
+    $this->assign('loggedUserID', CRM_Core_Session::getLoggedInContactID());
```

Comment:
Pretty sure this variable is already available to every smarty template, but whatever.

Reply:
Hmm - that would be good - but not sure I can see it anywhere